lib/sync/sync-scheduler.js

```diff
@@ -1,3 +1,140 @@
-module.exports = function() {
-  
-};
\ No newline at end of file
+var util = require('util');
+var _ = require('underscore');
+var async = require('async');
+var syncUtil = require('./util');
+var DatasetClient = require('./DatasetClient');
+var metrics = require('./sync-metrics');
+
+var metricsClient;
+var syncStorage;
+var syncLock;
+
+/**
+ * A scheduled job that will run to check the status of the datasets that are managed by the sync framework.
+ * @param {MongodbQueue} syncQueueImpl an instance of the MongodbQueue that is used to push sync requests
+ * @param {Object} options the configuration options for the scheduler
+ * @param {String} options.syncSchedulerLockName the name of the sync scheduler lock. Default: "locks:sync:SyncScheduler".
+ * @param {Number} options.timeBetweenChecks an interval that decides how often the scheduler should be running. In milliseconds. Default: 500.
+ * @param {Number} options.timeBeforeCrashAssumed a value that decides how long the lock should be accquired for. In milliseconds. Default: 20000.
+ */
+function SyncScheduler(syncQueueImpl, options) {
+  if (!syncQueueImpl) {
+    throw new Error('syncQueueImpl is required');
+  }
+  this.syncQueue = syncQueueImpl;
+  options = options || {};
+  this.syncSchedulerLockName = 'locks:sync:SyncScheduler' || options.syncSchedulerLockName;
+  this.timeBetweenChecks = options.timeBetweenChecks || 500;
+  this.timeBeforeCrashAssumed = options.timeBeforeCrashAssumed || 20000;
+}
+
+/**
+ * Loop through all the dataset clients and see if any of them is due to sync with the backend, or if the dataset is not active anymore and should be removed.
+ * @param {Function} cb the callback function
+ */
+SyncScheduler.prototype.checkDatasetsForSyncing = function(cb) {
+  var self = this;
+  var timer = metrics.startTimer();
+  syncStorage.listDatasetClients(function(err, datasetClients){
+    var timing;
```

Comment:
Do we need this? I think we can just use `timer.stop()` in the two places it's needed